test/inductor/test_cpu_repro.py

```diff
@@ -2237,8 +2190,10 @@ def fn(a):
         metrics.reset()
         x = torch.randn(1, 384, 20, 20).to(memory_format=torch.channels_last)
         self.common(fn, (x,))
-        assert metrics.generated_cpp_vec_kernel_count == 1
+        if codecache.valid_vec_isa_list():
+            assert metrics.generated_cpp_vec_kernel_count == 1
 
+    @requires_vectorization
     def test_non_contiguous_index_with_constant_stride(self):
```

Comment:
Is this necessary?

Reply:
Yes. This is one of the tests that fail on non-AVX systems. IIRC `FileCheck().check_count("cpp_fused", 2, exactly=True).run(code)` fails and would at least need to be guarded by `if codecache.valid_vec_isa_list():`

Reply:
This is weird. "cpp_fused" is from the kernel name which should be the same for vectorized and non-vectorized kernels. Can you share the error info and the generated code?

Reply:
True. Turns out, as I was testing 2.1.2 not the current 2.2-rc, the code has changed: https://github.com/pytorch/pytorch/commit/0a147fd11287f850538b8e4b75ed344ecbd1ace7#diff-042e454a3b5e4ae5aea2ff41066b5646ab068ca92b7e0adb5f9a81c6d7cec422

So this might indeed no longer be required.

Reply:
Removed this decorator